libevmasm/Assembly.h

```diff
@@ -174,6 +177,12 @@ class Assembly
 	std::map<util::h256, std::string> m_libraries; ///< Identifiers of libraries to be linked.
 	std::map<util::h256, std::string> m_immutables; ///< Identifiers of immutables.
 
+	/// map from sub object identifier to vector representing path to particular subobject
+	/// this map is used only for subobjects which are not direct subobjects (where path is having more than one value)
+	std::map<size_t, std::vector<size_t>> m_subObjectPaths;
+	/// current sub object id (this will be used for object which are not direct subobjects)
+	size_t m_currentSubObjectId = std::numeric_limits<size_t>::max();
```

Comment:
Isn't this just `m_subObjectPaths.size()`?

Reply:
Ah no, it's not - but it can be derived from it.

Reply:
Indeed, changing